openfe/setup/ligandatommapping.py

```diff
@@ -29,18 +30,23 @@ class LigandAtomMapping(Serializable):
     def to_dict(self):
         """Serialize to dict"""
         return {
-            'molA': self.molA.to_dict(),
-            'molB': self.molB.to_dict(),
+            # openff serialization doesn't go deep, so stringify at this level
+            'molA': self.molA.to_json(),
+            'molB': self.molB.to_json(),
             'molA_to_molB': self.molA_to_molB,
         }
 
     @classmethod
     def from_dict(cls, d: dict):
         """Deserialize from dict"""
+        # the mapping dict gets mangled sometimes
+        mapping = d['molA_to_molB']
+        fixed = {int(k): int(v) for k, v in mapping.items()}
```

Comment:
Huh, TLDR context for this mangling?

Reply:
not sure tbh, it was coming out as `{'1': 1, '2': 2...` so going through and really really making them ints seemed to fix that